### DISCUSSION TITLE
JSX parsing failure when mixing ts, tsx and js files

### DISCUSSION THREAD

**Discussion by hungify2022** [vitejs/vite-plugin-react-swc#65] [Ideas]
Firstly, thanks for the useful plugin.

**Context** 
My project is a bit of a mess.  `.js` and `.tsx`  for the component but I got an error

I am trying to migrate from Babel to Swc with Typescript but I have some issue

```
Failed to parse source for import analysis because the content contains invalid JS syntax. If you are using JSX, make sure to name the file with the .jsx or .tsx extension.
```

 I'm wondering if there's some setting `.swcrc` I can use to try to mix `.js` and '.tsx' files for now.

**Comment by ArnaudBarre** [comment]
Using JSX inside JS will not be supported by this plugin.
Rename your js files containing jsx to `.jsx` or `.tsx`.
More info: https://github.com/vitejs/vite-plugin-react-swc/issues/57#issuecomment-1425931254

**Reply by hungify2022** [reply]
Thanks!
